test/file.jl

```diff
@@ -36,6 +36,10 @@ end
     cd(pwd_)
 end
 
+# which (TODO: tests for Windows)
+@unix_only @test !isnull(which("ls"))
+@unix_only @test isfile(get(which("ls")))
+@unix_only @test isnull(which("abracadabra"))
```

Comment:
It might be OK for tests to fail if `ls` isn't on the path, but failing if `abracadabra` is on the path seems kind of wrong. You should find a more random non-existant path. Might I suggest something like `vyqMEOu1xfKCePhtZro7NUKbDmOEtx5ZmVGfUH`?